File: arpg_prg/Fantasy/Assets/Code/Core/NetWork/EnterRoomAction.cs
using System;

namespace Net
{
	public class EnterRoomAction : LogicAction<EnterRoomRequest, EnterRoomRespond>
	{
		public EnterRoomAction() : base(1)
		{
			
		}

		public override void OnRequest (EnterRoomRequest request, object userdata)
		{
			request.sessionID = 1111;
			request.roomID = 2222;
			request.playerID = 3333;

			request.list.Add (4444);
		}

		public override void OnRespond (EnterRoomRespond response, object userdata)
		{
			Console.WriteLine (response.roomID);
		}
	}
}

File: arpg_prg/server_prg/src/com/game/actions/EnterRoomAction.java
package com.game.actions;

import com.game.action.ids.ActionIDs;
import com.game.core.LogicAction;
import com.game.message.EnterRoomRequest;
import com.game.message.EnterRoomRespond;

public class EnterRoomAction extends LogicAction<EnterRoomRequest, EnterRoomRespond>
{
	public EnterRoomAction() 
	{
		super(ActionIDs.EnterRoomActionID, EnterRoomRequest.class, EnterRoomRespond.class);
	}

	@Override
	public void OnRequest(EnterRoomRequest request, Object userdata) 
	{
		System.out.println(request.sessionID);
		System.out.println(request.roomID);
		System.out.println(request.playerID);
		System.out.println(request.list.get(0));
	}

	@Override
	public void OnRespond(EnterRoomRespond response, Object userdadta) 
	{
		response.roomID = 111111111;
	}
	
}
